modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/rule-builder/RuleList/RuleList.es.js

```diff
@@ -0,0 +1,628 @@
+			return;
+		}
+
+		this.setState({
+			dropdownExpandedIndex: -1,
+		});
+	}
+
+	_handleDropdownClicked(event) {
+		event.preventDefault();
+
+		const {dropdownExpandedIndex} = this;
+		const ruleNode = event.delegateTarget.closest('.component-action');
+
+		let ruleIndex = parseInt(ruleNode.dataset.ruleIndex, 10);
+
+		if (ruleIndex === dropdownExpandedIndex) {
+			ruleIndex = -1;
+		}
+
+		this.setState({
+			dropdownExpandedIndex: ruleIndex,
+		});
+	}
+
+	_handleRuleCardClicked({data, target}) {
+		const cardElement = target.element.closest('[data-card-id]');
+		const cardId = parseInt(cardElement.getAttribute('data-card-id'), 10);
+
+		if (data.item.settingsItem == 'edit') {
+			this.emit('ruleEdited', {
+				ruleId: cardId,
+			});
+		}
+		else if (data.item.settingsItem == 'delete') {
+			if (
+				!data.item.confirm ||
+				confirm(
+					Liferay.Language.get(
+						'you-cannot-create-rules-with-nested-functions.-are-you-sure-you-want-to-delete-this-rule'
+					)
+				)
+			) {
+				this.emit('ruleDeleted', {
+					ruleId: cardId,
+				});
+			}
+		}
+	}
+
+	_hasNestedCondition(rule) {
+		return (
+			rule.conditions.find((condition) =>
+				condition.operands.find((operand) =>
+					operand.value.match(/[aA-zZ]+[(].*[,]+.*[)]/)
+				)
+			) !== undefined
+		);
+	}
+
+	_setDataProviderNames(states) {
+		const {rules} = states;
+
+		if (this.dataProvider) {
+			for (let rule = 0; rule < rules.length; rule++) {
+				const actions = rules[rule].actions;
+
+				actions.forEach((action) => {
+					if (action.action === 'auto-fill') {
+						const dataProviderName = this._getDataProviderName(
+							action.ddmDataProviderInstanceUUID
+						);
+
+						action.dataProviderName = dataProviderName;
+					}
+				});
+			}
+		}
+
+		return rules;
+	}
+}
+
+/**
+ * Prints the DDM form card rule.
+ */
+const EmptyList = ({message}) => {
+    return (
+        <div className="main-content-body">
+            <div className="sheet taglib-empty-result-message">
+                <div className="taglib-empty-result-message-header"></div>
+
+                {message && (
+                    <div className="sheet-text text-center text-muted">
+                        <p className="text-default">{message}</p>
+                    </div>
+                )}
+            </div>
+        </div>
+    )
+}
+
+/**
+ * Prints the DDM form card rule.
+ */
+const Label = ({content}) => {
+    return (
+        <span className="label label-lg label-secondary" data-original-title={content} title={content}>
+            <span className="text-truncate-inline">
+                <span className="text-truncate">{content}</span>
+            </span>
+	    </span>
+    )
+}
+
+
+/**
+ * Prints Rules Conditions.
+ */
+const Condition = ({
+    operandType,
+    operandLabel,
+    operandValue
+}) => {
+    const SubCondition = ({operandType}) => {
+        if(['double', 'integer', 'json', 'option', 'string', 'text', 'numeric'].includes(operandType)){
```

Comment:
Please, don't forget to answer https://github.com/liferay-forms/liferay-portal/pull/32 comments :)